File: packages/php/src/ConversionResult.php
<?php

namespace UmmAlQura;

class ConversionResult
{
    public function __construct(
        private GregorianDate|HijriDate $input,
        private GregorianDate|HijriDate $output,
        private int $jdn,
        private DayOfWeek $dayOfWeek,
        private string $locale,
        private string $libraryVersion,
        private string $dataVersion
    ) {}

    public function getInput(): GregorianDate|HijriDate { return $this->input; }
    public function getOutput(): GregorianDate|HijriDate { return $this->output; }
    public function getJdn(): int { return $this->jdn; }
    public function getDayOfWeek(): DayOfWeek { return $this->dayOfWeek; }
    public function getLocale(): string { return $this->locale; }
    public function getLibraryVersion(): string { return $this->libraryVersion; }
    public function getDataVersion(): string { return $this->dataVersion; }

    public function toArray(): array
    {
        return [
            'input' => $this->input->toArray(),
            'output' => $this->output->toArray(),
            'jdn' => $this->jdn,
            'day_of_week' => $this->dayOfWeek->toArray(),
            'locale' => $this->locale,
            'library_version' => $this->libraryVersion,
            'data_version' => $this->dataVersion,
        ];
    }
}

File: packages/java/src/main/java/com/ummalqura/ConversionResult.java
package com.ummalqura;

/**
 * Conversion result
 */
public class ConversionResult {
    private final Object input;
    private final Object output;
    private final int jdn;
    private final DayOfWeek dayOfWeek;
    private final String locale;
    private final String libraryVersion;
    private final String dataVersion;

    public ConversionResult(Object input, Object output, int jdn, DayOfWeek dayOfWeek,
                          String locale, String libraryVersion, String dataVersion) {
        this.input = input;
        this.output = output;
        this.jdn = jdn;
        this.dayOfWeek = dayOfWeek;
        this.locale = locale;
        this.libraryVersion = libraryVersion;
        this.dataVersion = dataVersion;
    }

    public Object getInput() { return input; }
    public Object getOutput() { return output; }
    public int getJdn() { return jdn; }
    public DayOfWeek getDayOfWeek() { return dayOfWeek; }
    public String getLocale() { return locale; }
    public String getLibraryVersion() { return libraryVersion; }
    public String getDataVersion() { return dataVersion; }
}
